cairo-contracts/packages/clients/src/cometbft/component.cairo

```diff
@@ -77,6 +78,30 @@ pub mod CometClientComponent {
             comet_client_state.latest_height
         }
 
+        fn update_height_before(
+            self: @ComponentState<TContractState>, client_sequence: u64, target_height: Height
+        ) -> Height {
+            let update_heights = self.read_update_heights(client_sequence);
```

Comment:
If we need to rebuild a whole array and then iterate over it, it might be better to just read directly from the storage, and then stop reading once we find the correct height.